src/porepy/utils/grid_utils.py

```diff
@@ -258,3 +252,72 @@ def switch_sign_if_inwards_normal(
 
     # Create the diagonal matrix.
     return sps.dia_matrix((sgn_mat, 0), shape=(sgn_mat.size, sgn_mat.size))
+
+
+def star_shape_cell_centers(g: "pp.Grid", as_nan: bool = False) -> np.ndarray:
+    """
+    For a given grid compute the star shape center for each cell.
+    The algorithm computes the half space intersections, by using the above method
+    half_space_pt,
+    of the spaces defined by the cell faces and the face normals.
+    This is a wrapper method that operate on a grid.
+
+    Parameters
+    ----------
+    g: pp.Grid
+        the grid
+    as_nan: bool, optional
+        Decide whether, in case some cells are not star-shaped return nan as
+        new center. Otherwise an exception is raised (default behaviour).
+
+    Returns
+    -------
+    np.ndarray
+        The new cell centers.
+
+    """
+
+    # no need for 1d or 0d grids
```

Comment:
```suggestion
    # Star-shaped cell centers are trivial for 1d or 0d grids. No need to compute them:
```

Reply:
But should we check whether the cell centers are correct, though? I guess the consistent thing would be mean of nodes in 1d and node coordinate in 0d.

Reply:
Not sure, to be honest. This is legacy code which seems to work for the interested users, so I'm inclined to leave it as is.